### PR TITLE
Refactor: Remove web_steps.rb

### PR DESCRIPTION
**Pull request by luke-hill**
## Summary

Remove the bane of existence

## Details

This half removes `web_steps.rb` from the codebase. Portions of it still remain (But only in test code) and there is additional refactor work to tidy up the tests more-over.

However this will remove the ability to generate web_steps using the generator e.t.c. So it is a breaking change.

## Motivation and Context

We've tried to get rid of them for ages. Now with a v2 cut as well, we can rip the plaster off.

## How Has This Been Tested?

CI

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by marnen** [comment]
I’ll be sorry to see it go (as I was sorry to see it removed years ago, even while agreeing with much of the reasoning). Although I think not generating web_steps anymore is a good decision, I usually wind up reimplementing about 2/3 of web_steps on any project I do.

**Comment by luke-hill** [comment]
I think the 2 key takeaways from the time it's been here is.

1) Re-implementing code as cukes is bad
2) Re-implementing several lines of code in 1 cuke is good

Other than that, it's been long enough so I'm content removing it as a v2.0 release sometime in the winter.

**Comment by luke-hill** [comment]
For now I'm just marking this as ready for review to see how it fares on CI. 

Locally this passes most stuff (I've not tested all the combinations)

**Comment by luke-hill** [comment]
This is now ready to review. And is passing CI

**Comment by marnen** [comment]
@luke-hill:

Was this a response to me?

> I think the 2 key takeaways from the time it's been here is.

> Re-implementing code as cukes is bad
> Re-implementing several lines of code in 1 cuke is good

If so, I can’t make any sense out of it despite 12 years of working with Cucumber. In particular I have no idea what a “cuke” is. From context, I had at first assumed you mean a Cucumber step definition, but then that still doesn’t make sense.

File: features/annotations.feature [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Feature: Annotations
 
-  I order to track my development progress
+  In order to track my development progress
```
Comment (olleolleolle) [inline features/annotations.feature]: Thanks for repairing this!

File: features/emulate_javascript.feature [hunk 2]
```diff
@@ -63,26 +71,33 @@ Feature: Emulate Javascript
...
+          And I am on the widgets page
           Then I should see "wrench"
           When I follow "Destroy"
           Then I should not see "denied"
           And I should be on the widgets page
           And I should not see "wrench"
       """
-    And I write to "features/step_definitions/widget_steps.rb" with:
+    And I append to "features/step_definitions/widget_steps.rb" with:
+      # TODO: Remove the newline below (Required) once bug is fixed: https://github.com/cucumber/aruba/issues/662
```
Comment (olleolleolle) [inline features/emulate_javascript.feature]: I appreciate the TODO comment here! ❤️

**Comment by luke-hill** [comment]
> @luke-hill:
> 
> Was this a response to me?
> 
> > I think the 2 key takeaways from the time it's been here is.
> 
> > Re-implementing code as cukes is bad
> > Re-implementing several lines of code in 1 cuke is good
> 
> If so, I can’t make any sense out of it despite 12 years of working with Cucumber. In particular I have no idea what a “cuke” is. From context, I had at first assumed you meant a Cucumber step definition, but then that still doesn’t make sense.

No sorry, this was just a general note to anyone who is watching the topic e.t.c.

**Pull request merged**